src/sentry/middleware/integrations/parsers/slack.py

```diff
@@ -1,11 +1,155 @@
+                return None
+            return Integration.objects.filter(id=slack_request.integration.id).first()
+
+        elif view_class_name in DJANGO_VIEW_ENDPOINTS:
+            # Parse the signed params to identify the associated integration
+            params = unsign(self.match.kwargs.get("signed_params"))
+            return Integration.objects.filter(id=params.get("integration_id")).first()
+
+    def get_response_from_first_region(self):
+        regions = self.get_regions_from_organizations()
+        first_region = regions[0]
+        response_map = self.get_responses_from_region_silos(regions=[first_region])
+        region_result = response_map[first_region.name]
+        if region_result.error is not None:
+            error = SiloClientError(region_result.error)
+            logger.error(
+                "region_error",
+                extra={"path": self.request.path, "region": first_region.name, "error": error},
+            )
+            # We want to fail loudly so that devs know this error happened on the region silo (for now)
+            raise SiloClientError(error)
+        return region_result.response
+
+    def get_response_from_all_regions(self):
+        regions = self.get_regions_from_organizations()
+        response_map = self.get_responses_from_region_silos(regions=regions)
+        successful_responses = [
+            result for result in response_map.values() if result.response is not None
+        ]
+        if len(successful_responses < 1):
+            error_map = {region: result.error for region, result in response_map.items()}
+            logger.error(
+                "all_regions_error",
+                extra={"path": self.request.path, "error_map": error_map},
+            )
+            raise SiloClientError("No successful region responses")
+        return successful_responses[0].response
+
+    def get_response(self):
+        """
+        Slack Webhook Requests all require synchronous responses.
+        """
+        view_class_name = self.match.func.view_class.__name__
+
+        regions = self.get_regions_from_organizations()
+        if len(regions) == 0:
+            logger.error("no_regions", extra={"path": self.request.path})
+            return self.get_response_from_control_silo()
+
+        view_class_name = self.match.func.view_class.__name__
+        if view_class_name in self.control_classes:
+            return self.get_response_from_control_silo()
+
+        if view_class_name == "SlackActionEndpoint":
+            drf_request: Request = SlackDMEndpoint().initialize_request(self.request)
+            slack_request = self.match.func.view_class.slack_request_class(drf_request)
+            action_option = SlackActionEndpoint.get_action_option(slack_request=slack_request)
+
+            # All actions other than those below are sent to every region
+            if action_option not in ["link", "ignore", "all_slack"]:
```

Comment:
We have this list defined a couple times. Should it have a consolidated definition so that it doesn't drift?